hypershift-operator/controllers/nodepool/inplace.go

```diff
@@ -169,7 +168,14 @@ func (r *NodePoolReconciler) reconcileMachineSet(ctx context.Context,
 		reason = hyperv1.NodePoolInplaceUpgradeFailedReason
 		message = machineSet.Annotations[nodePoolAnnotationUpgradeInProgressFalse]
 	}
-	if message != "" {
+
+	// Check if config needs to be updated.
+	isUpdatingConfig := isUpdatingConfig(nodePool, targetConfigHash)
+
+	// Check if version needs to be updated.
+	isUpdatingVersion := isUpdatingVersion(nodePool, targetVersion)
+
```

Comment:
thanks, this is a small improvement. Can we please follow up in another PR to:
Make sure the conditions are always set either false/true/unknown, regardless of message being empty or not.